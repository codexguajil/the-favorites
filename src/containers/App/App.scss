@font-face {
  font-family: 'helv-neue';
  src: url('./utils/assets/neue-helv/HelveticaNeue-BlackExt.woff') format('woff2');
}

@font-face {
  font-family: 'nav';
  src: url('./utils/assets/neue-helv/HelveticaNeue-HeavyItalic.woff') format('woff2');
}

@font-face {
  font-family: 'bebas';
  src: url('./utils/assets/bebas-neue/BebasNeue.otf') format('woff2');
}

* {
  margin: 0;
  padding: 0;
}

.App {
  background: url('./utils/assets/trexjpg');
  background-size: cover;
  text-align: center;
  height: 100%;
  width: 100%;
  margin: 0;
  padding: 0;
  overflow: scroll;
  z-index: -1;
}

header {
  padding: 25px;
}

h1 {
  color: hsla(0,0%,100%, 0.85);
  font-family: 'bebas';
  font-size: 6em;
  letter-spacing: 10px;
}

.nav {
  color: hsla(0,0%,100%, 0.85);
  font-family: 'nav';
  padding: 10px;
}