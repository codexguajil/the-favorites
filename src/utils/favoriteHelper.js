export const  validateFavorite = async (movie) => {
    const { user_id, id } = this.props
    const url = `http://localhost:3000/api/users/${user_id}/favorites`      
    if(user_id) {
      const allFavorites = await fetchData(url)
      allFavorites.data.forEach(favorite => {
        if(favorite.movie_id === id) {
          console.log('alreay in favorites')
        } else {
          this.addFavorite()
        }
      })
    } else {
      this.setState()
    }
  }

export const  addFavorite = async (e) => {
    const { id, title, rating, user_id, posterImage, synopsis, releaseDate } = this.props
    const url = "http://localhost:3000/api/users/favorites/new"
    const movie = {
      movie_id: id, 
      user_id: user_id, 
      title: title, 
      poster_path: posterImage, 
      release_date: releaseDate, 
      vote_average: rating, 
      overview: synopsis
    }
    try {
      await fetchData(url, {
        method: "POST", 
        body: JSON.stringify(movie),
        headers: {
          "Content-Type": "application/json"
        }
      })
    } catch(error) {
      console.log(error.message)
    }
  }