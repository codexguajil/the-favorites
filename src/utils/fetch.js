export const fetchData = async (url, options) => {
  const response = await fetch(url, options)
  if(response.ok) {
    const parsedData = response.json()
    return parsedData;
  } else {
    return 'error'
  }
} 