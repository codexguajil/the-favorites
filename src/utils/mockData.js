export const mockUser = {
  id: 1,
  name: "Taylor",
  password: "password",
  email: "tman2272@aol.com"
}

export const mockMovieResponse = {
page: 1,
results: [
  {adult: false,
  backdrop_path: "/hMANgfPHR1tRObNp2oPiOi9mMlz.jpg",
  genre_ids: [28, 12, 878],
  id: 424783,
  original_language: "en",
  original_title: "Bumblebee",
  overview: "On the run in the year 1987, Bumblebee finds refuge in a junkyard in a small Californian beach town. Charlie, on the cusp of turning 18 and trying to find her place in the world, discovers Bumblebee, battle-scarred and broken.  When Charlie revives him, she quickly learns this is no ordinary yellow VW bug.",
  popularity: 93.533,
  poster_path: "/fw02ONlDhrYjTSZV8XO6hhU3ds3.jpg",
  release_date: "2018-12-15",
  title: "Bumblebee",
  video: false,
  vote_average: 6.5,
  vote_count: 1414},
  {adult: false,
  backdrop_path: "/9QusGjxcYvfPD1THg6oW3RLeNn7.jpg",
  genre_ids:[28, 12, 14, 878],
  id: 297802,
  original_language: "en",
  original_title: "Aquaman",
  overview: "Once home to the most advanced civilization on Earth, Atlantis is now an underwater kingdom ruled by the power-hungry King Orm. With a vast army at his disposal, Orm plans to conquer the remaining oceanic people and then the surface world. Standing in his way is Arthur Curry, Orm's half-human, half-Atlantean brother and true heir to the throne.",
  popularity: 165.687,
  poster_path: "/5Kg76ldv7VxeX9YlcQXiowHgdX6.jpg",
  release_date: "2018-12-07",
  title: "Aquaman",
  video: false,
  vote_average: 6.8,
  vote_count: 4652,
  },
  {adult: false,
  backdrop_path: "/bTeRgkAavyw1eCtSkaww18wLYNP.jpg",
  genre_ids:[80, 18, 53],
  id: 504172,
  original_language: "en",
  original_title: "The Mule",
  overview: "Earl Stone, a man in his 80s who is broke, alone, and facing foreclosure of his business when he is offered a job that simply requires him to drive. Easy enough, but, unbeknownst to Earl, he’s just signed on as a drug courier for a Mexican cartel. He does so well that his cargo increases exponentially, and Earl hit the radar of hard-charging DEA agent Colin Bates.",
  popularity: 128.205,
  poster_path: "/oeZh7yEz3PMnZLgBPhrafFHRbVz.jpg",
  release_date: "2018-12-14",
  title: "The Mule",
  video: false,
  vote_average: 6.5,
  vote_count: 876},
  {adult: false,
  backdrop_path: "/uUiId6cG32JSRI6RyBQSvQtLjz2.jpg",
  genre_ids:[28, 12, 16, 878, 35],
  id: 324857,
  original_language: "en",
  original_title: "Spider-Man: Into the Spider-Verse",
  overview: "Miles Morales is juggling his life between being a high school student and being a spider-man. When Wilson Kingpin Fisk uses a super collider, others from across the Spider-Verse are transported to this dimension.",
  popularity: 157.433,
  poster_path: "/iiZZdoQBEYBv6id8su7ImL0oCbD.jpg",
  release_date: "2018-12-07",
  title: "Spider-Man: Into the Spider-Verse",
  video: false,
  vote_average: 8.5,
  vote_count: 2715},
  {backdrop_path: "/e7VzDMrYKXVrVon04Uqsrcgnf1k.jpg",
  first_air_date: "2019-03-15",
  genre_ids: [16, 10765],
  id: 86831,
  name: "Love, Death & Robots",
  origin_country: [],
  original_language: "en",
  original_name: "Love, Death & Robots",
  overview: "Terrifying creatures, wicked surprises and dark comedy converge in this NSFW anthology of animated stories presented by Tim Miller and David Fincher.",
  popularity: 161.862,
  poster_path: "/rlrRI2b6RvM9I9xOBTKqcTaehkE.jpg",
  vote_average: 8.6,
  vote_count: 29}],
  total_pages: 1000,
  total_results: 20000
}


export const mockDataResponse = [
  {id: 424783,
  posterImage: "https://image.tmdb.org/t/p/w500/fw02ONlDhrYjTSZV8XO6hhU3ds3.jpg",
  rating: 6.5,
  releaseDate: "2018-12-15",
  synopsis: "On the run in the year 1987, Bumblebee finds refuge in a junkyard in a small Californian beach town. Charlie, on the cusp of turning 18 and trying to find her place in the world, discovers Bumblebee, battle-scarred and broken.  When Charlie revives him, she quickly learns this is no ordinary yellow VW bug.",
  title: "Bumblebee"},
  {id: 297802,
  posterImage: "https://image.tmdb.org/t/p/w500/5Kg76ldv7VxeX9YlcQXiowHgdX6.jpg",
  rating: 6.8,
  releaseDate: "2018-12-07",
  synopsis: "Once home to the most advanced civilization on Earth, Atlantis is now an underwater kingdom ruled by the power-hungry King Orm. With a vast army at his disposal, Orm plans to conquer the remaining oceanic people and then the surface world. Standing in his way is Arthur Curry, Orm's half-human, half-Atlantean brother and true heir to the throne.",
  title: "Aquaman"},
  {id: 504172,
  posterImage: "https://image.tmdb.org/t/p/w500/oeZh7yEz3PMnZLgBPhrafFHRbVz.jpg",
  rating: 6.5,
  releaseDate: "2018-12-14",
  synopsis: "Earl Stone, a man in his 80s who is broke, alone, and facing foreclosure of his business when he is offered a job that simply requires him to drive. Easy enough, but, unbeknownst to Earl, he’s just signed on as a drug courier for a Mexican cartel. He does so well that his cargo increases exponentially, and Earl hit the radar of hard-charging DEA agent Colin Bates.",
  title: "The Mule"},
  {id: 399361,
  posterImage: "https://image.tmdb.org/t/p/w500/aBw8zYuAljVM1FeK5bZKITPH8ZD.jpg",
  rating: 6.3,
  releaseDate: "2019-03-06",
  synopsis: "Struggling to make ends meet, former special ops soldiers reunite for a high-stakes heist: stealing $75 million from a South American drug lord.",
  title: "Triple Frontier"},
  {id: 86831,
  posterImage: "https://image.tmdb.org/t/p/w500/rlrRI2b6RvM9I9xOBTKqcTaehkE.jpg",
  rating: 8.6,
  releaseDate: undefined,
  synopsis: "Terrifying creatures, wicked surprises and dark comedy converge in this NSFW anthology of animated stories presented by Tim Miller and David Fincher.",
  title: "Love, Death & Robots"}
]