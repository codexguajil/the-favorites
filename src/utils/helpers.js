import { fetchData } from './fetch';

export const cleanMovieData = (movies) => {
  return movies.results.map(movie => {
    return ({
        title: movie.title || movie.name,
        rating: movie.vote_average,
        id: movie.id,
        releaseDate: movie.release_date,
        synopsis: movie.overview,
        posterImage: movie.poster_path
    })
  })
}

export const fetchAllFavorites = async (id) => {
  const url = `http://localhost:3000/api/users/${id}/favorites`;
  const fetchedFavorites = await fetchData(url);
  const cleanedFavorites = cleanFavorites(fetchedFavorites.data)
  return cleanedFavorites;
}

const cleanFavorites = (favorites) => {
  return favorites.map(movie => {
    return movie.movie_id
  })
}