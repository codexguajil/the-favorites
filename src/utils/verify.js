import { fetchData } from './fetch';

export const verifyUser = (data, user) => {
  data.forEach(dataUser => {
    if(dataUser.email === user.email && dataUser.password === user.password) {
      console.log('match!')
    } else if(dataUser.email === user.email && dataUser.password !== user.password) {
      console.log('wrong password!')
    } else {
      console.log('user doesnt exist')
    }
  })
}

