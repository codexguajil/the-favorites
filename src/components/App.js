import React, { Component } from 'react';
import '../styles/App.css';
import { fetchData } from '../utils/fetch'
import { key } from '../utils/apiKEY';
import { cleanMovieData } from '../utils/helpers';
import { connect } from 'react-redux';
import { addAllMovies } from '../actions';
import MovieContainer from './MovieContainer';

class App extends Component {

  componentDidMount = () => {
    this.fetchMovies()
  }
  
  fetchMovies = async () => {
    const url = `https://api.themoviedb.org/3/trending/all/day?api_key=${key}`;
    const allMovies = await fetchData(url);
    const cleanData = cleanMovieData(allMovies);
    this.props.addAllMovies(cleanData);
  }

  render() {
    return (
      <div className="App">
        <header>
          <h1>Movie Tracker</h1>
        </header>
        <MovieContainer />
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  addAllMovies: (movies) => dispatch(addAllMovies(movies))
})

export default connect(null, mapDispatchToProps)(App);
