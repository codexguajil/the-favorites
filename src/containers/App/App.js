import React, { Component } from 'react';
// import './App.css';
import { fetchData } from '../../utils/fetch'
import { key } from '../../utils/apiKEY';
import { cleanMovieData } from '../../utils/helpers';
import { connect } from 'react-redux';
import { addAllMovies, addAllShows } from '../../actions';
import MovieContainer from '../MovieContainer';
import ShowsContainer from '../ShowsContainer';
import MovieDetails from '../../components/MovieDetails'
import { NavLink, Route } from 'react-router-dom';
import SignIn from '../SignIn';
import SignUp from '../SignUp';
import SignOut from '../SignOut';

class App extends Component {
  constructor() {
    super();
    this.state = {
      user: ''
    }
  }

  componentDidMount = () => {
    this.fetchMovies()
    this.fetchTv();
  }

  
  fetchMovies = async () => {
    let movies = [];
    for(let i = 1; i <= 3; i++) {
      let url = `https://api.themoviedb.org/3/trending/movie/day?api_key=${key}&page=${i}`;
      let allMovies = await fetchData(url);
      console.log(allMovies)
      let cleanData = cleanMovieData(allMovies);
      cleanData.forEach(movie => {
        movies.push(movie)
      })
    }
    this.props.addAllMovies(movies);
  }

  fetchTv = async () => {
    const url = `https://api.themoviedb.org/3/trending/tv/day?api_key=${key}`;
    const allShows = await fetchData(url);
    const cleanData = cleanMovieData(allShows);
    console.log(cleanData)
    this.props.addAllShows(cleanData);
  }

  render() {
    return (
      <div className="App">
        <header>
          <NavLink to='/login' className="nav">Log In</NavLink>
          <NavLink to='/signup' className="nav">Sign Up</NavLink>
          {/* <button onClick={this.fetchTv}>Show TV Shows</button> */}
          {
          this.props.user &&
            <div>
              <p>Welcome back!</p>
              <SignOut />
            </div>
          }
          <h1>MOVIE TRACKER</h1>
        </header>
        <h2 className="sub-header">Recommended Movies</h2>
        <Route exact path='/' component={MovieContainer} />
        <h2 className="sub-header">Recommended TV Shows</h2>
        <Route exact path='/' component={ShowsContainer} />
        <Route exact path='/login' component={SignIn} />
        <Route exact path='/signup' component={SignUp} />
        <Route path='/movies/:id' render={({ match }) => {
          const { id } = match.params
          const selectedMovie = this.props.movies.find(movie => {
            return movie.id == id
          })
          if(selectedMovie) {
            return <MovieDetails {...selectedMovie} />
          }
        }} />
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  addAllMovies: (movies) => dispatch(addAllMovies(movies)),
  addAllShows: (shows) => dispatch(addAllShows(shows))
})

const mapStateToProps = (state) => ({
  shows: state.shows,
  movies: state.movies,
  user: state.user.name
})

export default connect(mapStateToProps, mapDispatchToProps)(App);
