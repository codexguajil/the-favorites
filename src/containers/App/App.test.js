import React from 'react';
import App from './App';
import { shallow } from 'enzyme'
import { cleanMovieData } from '../../utils/helpers';
import { mockMovieResponse } from '../../utils/mockData';


describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <App />
    )
  })

  it('should match snapshot', () => {
    
    expect(wrapper).toMatchSnapshot();
  });

  describe('fetchMovies', () => {
    beforeEach(() => {
      window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
        ok: true,
        status: 200,
        json: () => Promise.resolve(mockMovieResponse)
      }))
    })
  })
})