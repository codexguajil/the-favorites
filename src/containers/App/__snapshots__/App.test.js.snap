// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className="App"
>
  <header>
    <Connect(Nav) />
    <h1>
      THE FAVORITES
    </h1>
    <Route
      component={[Function]}
      exact={true}
      path="/login"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/signup"
    />
  </header>
  <p
    className="message"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/favorites"
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    path="/details/:id"
    render={[Function]}
  />
</div>
`;

exports[`App should match snapshot when there is a selected movie 1`] = `
<div
  className="App"
>
  <header>
    <Connect(Nav) />
    <h1>
      THE FAVORITES
    </h1>
    <Route
      component={[Function]}
      exact={true}
      path="/login"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/signup"
    />
  </header>
  <p
    className="message"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/favorites"
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    path="/details/:id"
    render={[Function]}
  />
</div>
`;
